homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -257,45 +289,33 @@ async def async_step_pair(self, pair_info=None):
                 # PairVerify M4 - Device not recognised
                 # PairVerify M4 - Ed25519 signature verification failed
                 errors["pairing_code"] = "authentication_error"
+                self.finish_pairing = None
             except aiohomekit.UnknownError:
                 # An error occurred on the device whilst performing this
                 # operation.
                 errors["pairing_code"] = "unknown_error"
+                self.finish_pairing = None
             except aiohomekit.MaxPeersError:
                 # The device can't pair with any more accessories.
                 errors["pairing_code"] = "max_peers_error"
+                self.finish_pairing = None
             except aiohomekit.AccessoryNotFoundError:
                 # Can no longer find the device on the network
                 return self.async_abort(reason="accessory_not_found_error")
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Pairing attempt failed with an unhandled exception")
+                self.finish_pairing = None
                 errors["pairing_code"] = "pairing_failed"
 
-        discovery = await self.controller.find_ip_by_device_id(self.hkid)
-
-        try:
-            self.finish_pairing = await discovery.start_pairing(self.hkid)
-
-        except aiohomekit.BusyError:
-            # Already performing a pair setup operation with a different
-            # controller
-            errors["pairing_code"] = "busy_error"
-        except aiohomekit.MaxTriesError:
-            # The accessory has received more than 100 unsuccessful auth
-            # attempts.
-            errors["pairing_code"] = "max_tries_error"
-        except aiohomekit.UnavailableError:
-            # The accessory is already paired - cannot try to pair again.
-            return self.async_abort(reason="already_paired")
-        except aiohomekit.AccessoryNotFoundError:
-            # Can no longer find the device on the network
-            return self.async_abort(reason="accessory_not_found_error")
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Pairing attempt failed with an unhandled exception")
-            errors["pairing_code"] = "pairing_failed"
+        if errors and "base" in errors:
+            return self.async_show_form(step_id="try_pair_later", errors=errors)
```

Comment:
We don't want to move to another step's form directly or pass errors between steps. Errors are step specific. Please move to the other step by awaiting the step method instead.

Reply:
Well, I had that way before, but I took it back to show form because it was less code 🙈 

I'll adjust in a new PR.

Reply:
https://github.com/home-assistant/core/pull/38742